v2/spiffetls/option.go

```diff
@@ -3,22 +3,42 @@ package spiffetls
 import (
 	"crypto/tls"
 	"net"
+
+	"github.com/zeebo/errs"
 )
 
+var spiffetlsErr = errs.Class("spiffetls")
+
 // DialOption is an option for dialing. Option's are also DialOption's.
-type DialOption interface{}
+type DialOption interface {
```

Comment:
nit: this diverges a little from the other patterns we've used in the codebase. I might suggest consistency, i.e. something like:
```
type DialOption interface{
    apply(*dialConfig)
}

type dialOption func(*dialConfig)

func (fn dialOption) apply(c *dialConfig) {
    fn(c)
}

func WithDialTLSConfigBase(base *tls.Config) DialOption {
    return dialOption(func(c *dialConfig) {
        c.baseTLSConf = base
    })
}
```